### PR TITLE
feat: add provider-url support to lease commands for on-chain independence

### PR DESCRIPTION
**Pull request by cloud-j-luna**
Adds provider URL support across lease-related commands, introduces a new provider-url flag and helper-based client construction, migrates provider client creation to option-based WithQueryClient, updates tests accordingly, adjusts Make targets to include provider-url variants, and bumps akash-api to v0.0.84.

[API PR](https://github.com/akash-network/akash-api/pull/229)

## How to test

```bash
$ make provider-lease-shell
provider-services lease-shell \
        --dseq      "1"        \
        --gseq      "1"        \
        --oseq      "1"        \
        --from      "main"    \
        --provider  "akash19mp92vqx6vu44uldmumtyjceladnmzegk94uf5" \
        --auth-type=jwt \
        web \
        date
Wed Sep 24 10:26:51 UTC 2025

$ make provider-url-lease-shell
provider-services lease-shell \
        --dseq      "1"        \
        --gseq      "1"        \
        --oseq      "1"        \
        --from      "main"    \
        --provider  "akash19mp92vqx6vu44uldmumtyjceladnmzegk94uf5" \
        --provider-url "https://localhost:8443" \
        --auth-type=jwt \
        web \
        date
Wed Sep 24 10:26:56 UTC 2025

##### NODE GOES OFFLINE ######

$ make provider-url-lease-shell
provider-services lease-shell \
        --dseq      "1"        \
        --gseq      "1"        \
        --oseq      "1"        \
        --from      "main"    \
        --provider  "akash19mp92vqx6vu44uldmumtyjceladnmzegk94uf5" \
        --provider-url "https://localhost:8443" \
        --auth-type=jwt \
        web \
        date
Wed Sep 24 10:27:01 UTC 2025

$ make provider-lease-shell
provider-services lease-shell \
        --dseq      "1"        \
        --gseq      "1"        \
        --oseq      "1"        \
        --from      "main"    \
        --provider  "akash19mp92vqx6vu44uldmumtyjceladnmzegk94uf5" \
        --auth-type=jwt \
        web \
        date
Error: post failed: Post "http://localhost:26657": dial tcp [::1]:26657: connect: connection refused
make: *** [../common-commands.mk:346: provider-lease-shell] Error 1
```

### CODE REVIEW

File: _run/kube/deployment.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ services:
         to:
           - global: true
   bew:
-    image: quay.io/ovrclk/demo-app
+    image: nginxdemos/hello:latest
```
Comment (cloud-j-luna) [inline _run/kube/deployment.yaml]: Changing these so that example deployment has logs to fetch

File: _run/common-commands.mk [hunk 2]
```diff
@@ -290,6 +290,68 @@ provider-lease-logs:
...
+provider-url-lease-logs:
+	$(PROVIDER_SERVICES) lease-logs \
+		-f \
+		--service="$(LEASE_SERVICES)" \
+		--dseq      "$(DSEQ)"     \
+		--gseq      "$(GSEQ)"        \
+		--oseq      "$(OSEQ)"        \
+		--from      "$(KEY_NAME)" \
+		--provider  "$(PROVIDER_ADDRESS)" \
+		--provider-url "https://localhost:8443" \
```
Comment (troian) [inline _run/common-commands.mk]: lets instead set AP_PROVIDER_URL once.
also, it should not be needed by default, i.e. this is edge case we test

File: _run/kube/deployment.yaml [hunk 3]
```diff
@@ -3,7 +3,7 @@ version: "2.0"
 
 services:
   web:
-    image: quay.io/ovrclk/demo-app
+    image: nginxdemos/hello:latest
```
Comment (troian) [inline _run/kube/deployment.yaml]: this may break e2e tests, quay.io/ovrclk/demo-app has endpoint which simulates deployment crash

File: cmd/provider-services/cmd/helpers.go [hunk 4]
```diff
@@ -15,6 +15,7 @@ import (
 	sdkclient "github.com/cosmos/cosmos-sdk/client"
 	"github.com/cosmos/cosmos-sdk/client/flags"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	pkgerrors "github.com/pkg/errors"
```
Comment (troian) [inline cmd/provider-services/cmd/helpers.go]: no github.com/pkg/errors please, it has been deprecated and we removed it in upcoming release

File: cmd/provider-services/cmd/helpers.go [hunk 5]
```diff
@@ -223,3 +225,66 @@ func loadAuthOpts(ctx context.Context, cctx sdkclient.Context, flags *pflag.Flag
...
+			return pkgerrors.Errorf("%s flag is required when using provider-url", req.name)
+		}
+	}
+
+	return nil
+}
+
+// constructLeaseIDFromProviderURL constructs a lease ID from flags when using provider URL
+// This bypasses RPC discovery and constructs the lease ID directly from command line flags
+func constructLeaseIDFromProviderURL(flags *pflag.FlagSet, owner string) (mtypes.LeaseID, error) {
```
Comment (troian) [inline cmd/provider-services/cmd/helpers.go]: leaseIDFromFlags

File: cmd/provider-services/cmd/helpers.go [hunk 6]
```diff
@@ -223,3 +225,66 @@ func loadAuthOpts(ctx context.Context, cctx sdkclient.Context, flags *pflag.Flag
...
+	}
+
+	return leaseID, nil
+}
+
+// createProviderClientFromFlags creates an akash provider client with appropriate options
+// based on whether a provider URL is specified via flags or not.
+// If provider-url flag is set, it uses WithProviderURL option.
+// Otherwise, it uses WithQueryClient option for blockchain queries.
+func createProviderClientFromFlags(
```
Comment (troian) [inline cmd/provider-services/cmd/helpers.go]: providerClientFromFlags

File: cmd/provider-services/cmd/leaseEvents.go [hunk 7]
```diff
@@ -43,22 +45,39 @@ func doLeaseEvents(cmd *cobra.Command) error {
...
+	var leases []mtypes.LeaseID
+	var cl qclient.QueryClient
 
-	leases, err := leasesForDeployment(cmd.Context(), cl, cmd.Flags(), dtypes.DeploymentID{
-		Owner: cctx.GetFromAddress().String(),
-		DSeq:  dseq,
-	})
-	if err != nil {
-		return markRPCServerError(err)
+	if providerURL != "" {
```
Comment (troian) [inline cmd/provider-services/cmd/leaseEvents.go]: this code is duplicated in many places

File: cmd/provider-services/cmd/leaseLogs.go [hunk 8]
```diff
@@ -145,6 +165,7 @@ func doLeaseLogs(cmd *cobra.Command) error {
 
 	for _, stream := range streams {
 		if stream.error != nil {
+			fmt.Println(stream.error)
```
Comment (troian) [inline cmd/provider-services/cmd/leaseLogs.go]: errors should be printed into stderr

File: _run/common-commands.mk (hunk 2 above)
Comment (cloud-j-luna) [reply _run/common-commands.mk]: Made the change. It is not needed by default this targets are specific to use this directly.

File: cmd/provider-services/cmd/helpers.go (hunk 5 above)
Comment (cloud-j-luna) [reply cmd/provider-services/cmd/helpers.go]: Already exists. Renaming to: `leaseIDWhenProviderURLIsProvided`.

File: cmd/provider-services/cmd/helpers.go (hunk 4 above)
Comment (cloud-j-luna) [reply cmd/provider-services/cmd/helpers.go]: Indeed, shouldn't have been there. Replaced with `errors`.

File: cmd/provider-services/cmd/leaseEvents.go (hunk 7 above)
Comment (cloud-j-luna) [reply cmd/provider-services/cmd/leaseEvents.go]: The checking conditions are the same on those commands. Either a non-empty provider URL is provided and we do  the *off-chain* connection or else we go *on-chain*. Code is similar but not the same exactly the same which makes extracting a function to reduce duplicated work trickier and might just end up with more code. I could move `providerURL != ""` to its own function for readability if that is a concern here.

All other comments were addressed.

File: _run/common-commands.mk [hunk 9]
```diff
@@ -290,6 +291,68 @@ provider-lease-logs:
...
+provider-url-lease-logs:
+	$(PROVIDER_SERVICES) lease-logs \
+		-f \
+		--service="$(LEASE_SERVICES)" \
+		--dseq      "$(DSEQ)"     \
+		--gseq      "$(GSEQ)"        \
+		--oseq      "$(OSEQ)"        \
+		--from      "$(KEY_NAME)" \
+		--provider  "$(PROVIDER_ADDRESS)" \
+		--provider-url "$(AP_PROVIDER_URL)" \
```
Comment (troian) [inline _run/common-commands.mk]: ✂️

File: _run/common-commands.mk [hunk 10]
```diff
@@ -15,6 +15,7 @@ CERT_HOSTNAME  ?= localhost
 LEASE_SERVICES ?= web
 
 RESOURCE_SERVER_HOST ?= localhost:8445
+AP_PROVIDER_URL      ?= https://localhost:8443
```
Comment (troian) [inline _run/common-commands.mk]: `AP_PROVIDER_URL        ?=`
